specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2022-10-31-preview/iotcentral.json

```diff
@@ -0,0 +1,9326 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Destination"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next page of destinations.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "DestinationExport": {
+      "description": "The destination export definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Export"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "transform": {
+          "type": "string",
+          "description": "Query for transforming the message structure to a particular output."
+        }
+      }
+    },
+    "DestinationReference": {
+      "description": "The destination reference definition.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the destination where data should be sent."
+        },
+        "transform": {
+          "type": "string",
+          "description": "Query for transforming the message structure to a particular output."
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "Device": {
+      "description": "The device definition.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique ID of the device.",
+          "type": "string",
+          "readOnly": true
+        },
+        "etag": {
+          "description": "ETag used to prevent conflict in device updates.",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Display name of the device.",
+          "type": "string"
+        },
+        "template": {
+          "description": "The device template definition for the device.",
+          "type": "string"
+        },
+        "enabled": {
+          "description": "Whether the device connection to IoT Central has been enabled.",
+          "type": "boolean"
+        },
+        "provisioned": {
+          "description": "Whether resources have been allocated for the device.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "simulated": {
+          "description": "Whether the device is simulated.",
+          "type": "boolean"
+        },
+        "organizations": {
+          "description": "List of organization IDs that the device is a part of, only one organization is supported today, multiple organizations will be supported soon.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "type": {
+          "description": "The type of the device.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "enum": [
+              "iotEdge"
+            ],
+            "x-ms-enum": {
+              "name": "DeviceType",
+              "modelAsString": false,
```

Comment:
You must be 100% sure you'll never support any other values. But if that's the case, what's the value of this `type` property at all? Seems this should be `modelAsString: true`. By default, you should assume `true`.

Reply:
I see `modelAsString: false` many times in models below. This seems implausible. You have it, for example, on an ISO8601 duration property with limited enum values. So what's the point of using ISO 8601 durations (which most languages don't support) if users can't specify their own, meaning it should be modelAsString: true.

Reply:
Agreed, I have updated the deviceType "modelAsString" to "true", as our team never release any SDKs, didn't realized that ISO8601 can't be used in other languages, I'd like to learn more on this, do you have any docs or references? I will update the modelAsString to true as you suggested.